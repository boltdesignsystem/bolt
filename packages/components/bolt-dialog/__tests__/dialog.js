// Refer to https://github.com/boltdesignsystem/bolt/wiki/Jest-Test:-Example-Jest-Test for more testing examples
import { render, stopServer } from '../../../testing/testing-helpers';
import schema from '../dialog.schema';
const { width, spacing, persistent, theme } = schema.properties;
let page, fixtures;

afterAll(async () => {
  await stopServer();
  await page.close();
});

beforeEach(async () => {
  await page.evaluate(() => {
    document.body.innerHTML = '';
  });
});

beforeAll(async () => {
  page = await global.__BROWSER__.newPage();
  await page.goto('http://127.0.0.1:4444/', {
    timeout: 0,
  });

  const dialogHeader = await render(
    '@bolt-components-dialog/dialog-header.twig',
    { content: 'Header content' },
  );
  const dialogBody = await render('@bolt-components-dialog/dialog-body.twig', {
    content: 'Body content',
  });
  const dialogFooter = await render(
    '@bolt-components-dialog/dialog-footer.twig',
    { content: 'Footer content' },
  );

  const defaultData = {
    content: [dialogHeader.html, dialogBody.html, dialogFooter.html].join(''),
  };

  fixtures = {
    defaultData,
  };
});

describe('Bolt Dialog component', () => {
  test('default', async () => {
    const results = await render('@bolt-components-dialog/dialog.twig', {
      ...fixtures.defaultData,
    });

    await expect(results.ok).toBe(true);
    await expect(results.html).toMatchSnapshot();
  });

  width.enum.forEach(async option => {
    test(`dialog width: ${option}`, async () => {
      const results = await render('@bolt-components-dialog/dialog.twig', {
        ...fixtures.defaultData,
        width: option,
      });
      expect(results.ok).toBe(true);
      expect(results.html).toMatchSnapshot();
    });
  });

  spacing.enum.forEach(async option => {
    test(`dialog spacing: ${option}`, async () => {
      const results = await render('@bolt-components-dialog/dialog.twig', {
        ...fixtures.defaultData,
        spacing: option,
      });
      expect(results.ok).toBe(true);
      expect(results.html).toMatchSnapshot();
    });
  });

  theme.enum.forEach(async option => {
    test(`dialog theme: ${option}`, async () => {
      const results = await render('@bolt-components-dialog/dialog.twig', {
        ...fixtures.defaultData,
        theme: option,
      });
      expect(results.ok).toBe(true);
      expect(results.html).toMatchSnapshot();
    });
  });

  test(`dialog persistent: true`, async () => {
    const results = await render('@bolt-components-dialog/dialog.twig', {
      ...fixtures.defaultData,
      persistent: true,
    });
    expect(results.ok).toBe(true);
    expect(results.html).toMatchSnapshot();
  });
});
