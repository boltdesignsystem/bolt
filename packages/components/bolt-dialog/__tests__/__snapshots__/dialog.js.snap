// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bolt Dialog component default 1`] = `
<dialog class="c-bolt-dialog c-bolt-dialog--hidden c-bolt-dialog__width-optimal c-bolt-dialog__spacing-medium t-bolt-xlight">
  <div class="c-bolt-dialog__close">
    <button type="button"
            data-dialog-close
            aria-label="Close dialog"
            class="e-bolt-button e-bolt-button--transparent e-bolt-button--icon-only"
    >
      <span class="e-bolt-button__icon-center"
            aria-hidden="true"
      >
        <bolt-icon name="close"
                   aria-hidden="true"
        >
          <span class="c-bolt-icon c-bolt-icon--close">
            <svg class="c-bolt-icon__icon"
                 xmlns="http://www.w3.org/2000/svg"
                 viewbox="0 0 24 24"
            >
              <path fill="var(--bolt-theme-icon, currentColor)"
                    fill-rule="evenodd"
                    d="M13.4 12l5.3-5.3c.4-.4.4-1 0-1.4a1 1 0 00-1.4 0L12 10.6 6.7 5.3a1 1 0 00-1.4 0 1 1 0 000 1.4l5.3 5.3-5.3 5.3A1 1 0 006 19c.3 0 .5-.1.7-.3l5.3-5.3 5.3 5.3a1 1 0 101.4-1.4L13.4 12z"
              >
              </path>
            </svg>
          </span>
        </bolt-icon>
      </span>
    </button>
  </div>
  <header class="c-bolt-dialog__header">
    <div class="c-bolt-dialog__header-content">
      Header content
    </div>
  </header>
  <div class="c-bolt-dialog__body">
    <article>
      Body content
    </article>
  </div>
  <div class="c-bolt-dialog__footer">
    Footer content
  </div>
</dialog>
`;

exports[`Bolt Dialog component dialog persistent: true 1`] = `
<dialog class="c-bolt-dialog c-bolt-dialog--hidden c-bolt-dialog__width-optimal c-bolt-dialog__spacing-medium t-bolt-xlight"
        data-dialog-persistent
>
  <header class="c-bolt-dialog__header">
    <div class="c-bolt-dialog__header-content">
      Header content
    </div>
  </header>
  <div class="c-bolt-dialog__body">
    <article>
      Body content
    </article>
  </div>
  <div class="c-bolt-dialog__footer">
    Footer content
  </div>
</dialog>
`;

exports[`Bolt Dialog component dialog spacing: large 1`] = `
<dialog class="c-bolt-dialog c-bolt-dialog--hidden c-bolt-dialog__width-optimal c-bolt-dialog__spacing-large t-bolt-xlight">
  <div class="c-bolt-dialog__close">
    <button type="button"
            data-dialog-close
            aria-label="Close dialog"
            class="e-bolt-button e-bolt-button--transparent e-bolt-button--icon-only"
    >
      <span class="e-bolt-button__icon-center"
            aria-hidden="true"
      >
        <bolt-icon name="close"
                   aria-hidden="true"
        >
          <span class="c-bolt-icon c-bolt-icon--close">
            <svg class="c-bolt-icon__icon"
                 xmlns="http://www.w3.org/2000/svg"
                 viewbox="0 0 24 24"
            >
              <path fill="var(--bolt-theme-icon, currentColor)"
                    fill-rule="evenodd"
                    d="M13.4 12l5.3-5.3c.4-.4.4-1 0-1.4a1 1 0 00-1.4 0L12 10.6 6.7 5.3a1 1 0 00-1.4 0 1 1 0 000 1.4l5.3 5.3-5.3 5.3A1 1 0 006 19c.3 0 .5-.1.7-.3l5.3-5.3 5.3 5.3a1 1 0 101.4-1.4L13.4 12z"
              >
              </path>
            </svg>
          </span>
        </bolt-icon>
      </span>
    </button>
  </div>
  <header class="c-bolt-dialog__header">
    <div class="c-bolt-dialog__header-content">
      Header content
    </div>
  </header>
  <div class="c-bolt-dialog__body">
    <article>
      Body content
    </article>
  </div>
  <div class="c-bolt-dialog__footer">
    Footer content
  </div>
</dialog>
`;

exports[`Bolt Dialog component dialog spacing: medium 1`] = `
<dialog class="c-bolt-dialog c-bolt-dialog--hidden c-bolt-dialog__width-optimal c-bolt-dialog__spacing-medium t-bolt-xlight">
  <div class="c-bolt-dialog__close">
    <button type="button"
            data-dialog-close
            aria-label="Close dialog"
            class="e-bolt-button e-bolt-button--transparent e-bolt-button--icon-only"
    >
      <span class="e-bolt-button__icon-center"
            aria-hidden="true"
      >
        <bolt-icon name="close"
                   aria-hidden="true"
        >
          <span class="c-bolt-icon c-bolt-icon--close">
            <svg class="c-bolt-icon__icon"
                 xmlns="http://www.w3.org/2000/svg"
                 viewbox="0 0 24 24"
            >
              <path fill="var(--bolt-theme-icon, currentColor)"
                    fill-rule="evenodd"
                    d="M13.4 12l5.3-5.3c.4-.4.4-1 0-1.4a1 1 0 00-1.4 0L12 10.6 6.7 5.3a1 1 0 00-1.4 0 1 1 0 000 1.4l5.3 5.3-5.3 5.3A1 1 0 006 19c.3 0 .5-.1.7-.3l5.3-5.3 5.3 5.3a1 1 0 101.4-1.4L13.4 12z"
              >
              </path>
            </svg>
          </span>
        </bolt-icon>
      </span>
    </button>
  </div>
  <header class="c-bolt-dialog__header">
    <div class="c-bolt-dialog__header-content">
      Header content
    </div>
  </header>
  <div class="c-bolt-dialog__body">
    <article>
      Body content
    </article>
  </div>
  <div class="c-bolt-dialog__footer">
    Footer content
  </div>
</dialog>
`;

exports[`Bolt Dialog component dialog spacing: none 1`] = `
<dialog class="c-bolt-dialog c-bolt-dialog--hidden c-bolt-dialog__width-optimal c-bolt-dialog__spacing-none t-bolt-xlight">
  <div class="c-bolt-dialog__close">
    <button type="button"
            data-dialog-close
            aria-label="Close dialog"
            class="e-bolt-button e-bolt-button--transparent e-bolt-button--icon-only"
    >
      <span class="e-bolt-button__icon-center"
            aria-hidden="true"
      >
        <bolt-icon name="close"
                   aria-hidden="true"
        >
          <span class="c-bolt-icon c-bolt-icon--close">
            <svg class="c-bolt-icon__icon"
                 xmlns="http://www.w3.org/2000/svg"
                 viewbox="0 0 24 24"
            >
              <path fill="var(--bolt-theme-icon, currentColor)"
                    fill-rule="evenodd"
                    d="M13.4 12l5.3-5.3c.4-.4.4-1 0-1.4a1 1 0 00-1.4 0L12 10.6 6.7 5.3a1 1 0 00-1.4 0 1 1 0 000 1.4l5.3 5.3-5.3 5.3A1 1 0 006 19c.3 0 .5-.1.7-.3l5.3-5.3 5.3 5.3a1 1 0 101.4-1.4L13.4 12z"
              >
              </path>
            </svg>
          </span>
        </bolt-icon>
      </span>
    </button>
  </div>
  <header class="c-bolt-dialog__header">
    <div class="c-bolt-dialog__header-content">
      Header content
    </div>
  </header>
  <div class="c-bolt-dialog__body">
    <article>
      Body content
    </article>
  </div>
  <div class="c-bolt-dialog__footer">
    Footer content
  </div>
</dialog>
`;

exports[`Bolt Dialog component dialog spacing: small 1`] = `
<dialog class="c-bolt-dialog c-bolt-dialog--hidden c-bolt-dialog__width-optimal c-bolt-dialog__spacing-small t-bolt-xlight">
  <div class="c-bolt-dialog__close">
    <button type="button"
            data-dialog-close
            aria-label="Close dialog"
            class="e-bolt-button e-bolt-button--transparent e-bolt-button--icon-only"
    >
      <span class="e-bolt-button__icon-center"
            aria-hidden="true"
      >
        <bolt-icon name="close"
                   aria-hidden="true"
        >
          <span class="c-bolt-icon c-bolt-icon--close">
            <svg class="c-bolt-icon__icon"
                 xmlns="http://www.w3.org/2000/svg"
                 viewbox="0 0 24 24"
            >
              <path fill="var(--bolt-theme-icon, currentColor)"
                    fill-rule="evenodd"
                    d="M13.4 12l5.3-5.3c.4-.4.4-1 0-1.4a1 1 0 00-1.4 0L12 10.6 6.7 5.3a1 1 0 00-1.4 0 1 1 0 000 1.4l5.3 5.3-5.3 5.3A1 1 0 006 19c.3 0 .5-.1.7-.3l5.3-5.3 5.3 5.3a1 1 0 101.4-1.4L13.4 12z"
              >
              </path>
            </svg>
          </span>
        </bolt-icon>
      </span>
    </button>
  </div>
  <header class="c-bolt-dialog__header">
    <div class="c-bolt-dialog__header-content">
      Header content
    </div>
  </header>
  <div class="c-bolt-dialog__body">
    <article>
      Body content
    </article>
  </div>
  <div class="c-bolt-dialog__footer">
    Footer content
  </div>
</dialog>
`;

exports[`Bolt Dialog component dialog theme: dark 1`] = `
<dialog class="c-bolt-dialog c-bolt-dialog--hidden c-bolt-dialog__width-optimal c-bolt-dialog__spacing-medium t-bolt-dark">
  <div class="c-bolt-dialog__close">
    <button type="button"
            data-dialog-close
            aria-label="Close dialog"
            class="e-bolt-button e-bolt-button--transparent e-bolt-button--icon-only"
    >
      <span class="e-bolt-button__icon-center"
            aria-hidden="true"
      >
        <bolt-icon name="close"
                   aria-hidden="true"
        >
          <span class="c-bolt-icon c-bolt-icon--close">
            <svg class="c-bolt-icon__icon"
                 xmlns="http://www.w3.org/2000/svg"
                 viewbox="0 0 24 24"
            >
              <path fill="var(--bolt-theme-icon, currentColor)"
                    fill-rule="evenodd"
                    d="M13.4 12l5.3-5.3c.4-.4.4-1 0-1.4a1 1 0 00-1.4 0L12 10.6 6.7 5.3a1 1 0 00-1.4 0 1 1 0 000 1.4l5.3 5.3-5.3 5.3A1 1 0 006 19c.3 0 .5-.1.7-.3l5.3-5.3 5.3 5.3a1 1 0 101.4-1.4L13.4 12z"
              >
              </path>
            </svg>
          </span>
        </bolt-icon>
      </span>
    </button>
  </div>
  <header class="c-bolt-dialog__header">
    <div class="c-bolt-dialog__header-content">
      Header content
    </div>
  </header>
  <div class="c-bolt-dialog__body">
    <article>
      Body content
    </article>
  </div>
  <div class="c-bolt-dialog__footer">
    Footer content
  </div>
</dialog>
`;

exports[`Bolt Dialog component dialog theme: light 1`] = `
<dialog class="c-bolt-dialog c-bolt-dialog--hidden c-bolt-dialog__width-optimal c-bolt-dialog__spacing-medium t-bolt-light">
  <div class="c-bolt-dialog__close">
    <button type="button"
            data-dialog-close
            aria-label="Close dialog"
            class="e-bolt-button e-bolt-button--transparent e-bolt-button--icon-only"
    >
      <span class="e-bolt-button__icon-center"
            aria-hidden="true"
      >
        <bolt-icon name="close"
                   aria-hidden="true"
        >
          <span class="c-bolt-icon c-bolt-icon--close">
            <svg class="c-bolt-icon__icon"
                 xmlns="http://www.w3.org/2000/svg"
                 viewbox="0 0 24 24"
            >
              <path fill="var(--bolt-theme-icon, currentColor)"
                    fill-rule="evenodd"
                    d="M13.4 12l5.3-5.3c.4-.4.4-1 0-1.4a1 1 0 00-1.4 0L12 10.6 6.7 5.3a1 1 0 00-1.4 0 1 1 0 000 1.4l5.3 5.3-5.3 5.3A1 1 0 006 19c.3 0 .5-.1.7-.3l5.3-5.3 5.3 5.3a1 1 0 101.4-1.4L13.4 12z"
              >
              </path>
            </svg>
          </span>
        </bolt-icon>
      </span>
    </button>
  </div>
  <header class="c-bolt-dialog__header">
    <div class="c-bolt-dialog__header-content">
      Header content
    </div>
  </header>
  <div class="c-bolt-dialog__body">
    <article>
      Body content
    </article>
  </div>
  <div class="c-bolt-dialog__footer">
    Footer content
  </div>
</dialog>
`;

exports[`Bolt Dialog component dialog theme: xdark 1`] = `
<dialog class="c-bolt-dialog c-bolt-dialog--hidden c-bolt-dialog__width-optimal c-bolt-dialog__spacing-medium t-bolt-xdark">
  <div class="c-bolt-dialog__close">
    <button type="button"
            data-dialog-close
            aria-label="Close dialog"
            class="e-bolt-button e-bolt-button--transparent e-bolt-button--icon-only"
    >
      <span class="e-bolt-button__icon-center"
            aria-hidden="true"
      >
        <bolt-icon name="close"
                   aria-hidden="true"
        >
          <span class="c-bolt-icon c-bolt-icon--close">
            <svg class="c-bolt-icon__icon"
                 xmlns="http://www.w3.org/2000/svg"
                 viewbox="0 0 24 24"
            >
              <path fill="var(--bolt-theme-icon, currentColor)"
                    fill-rule="evenodd"
                    d="M13.4 12l5.3-5.3c.4-.4.4-1 0-1.4a1 1 0 00-1.4 0L12 10.6 6.7 5.3a1 1 0 00-1.4 0 1 1 0 000 1.4l5.3 5.3-5.3 5.3A1 1 0 006 19c.3 0 .5-.1.7-.3l5.3-5.3 5.3 5.3a1 1 0 101.4-1.4L13.4 12z"
              >
              </path>
            </svg>
          </span>
        </bolt-icon>
      </span>
    </button>
  </div>
  <header class="c-bolt-dialog__header">
    <div class="c-bolt-dialog__header-content">
      Header content
    </div>
  </header>
  <div class="c-bolt-dialog__body">
    <article>
      Body content
    </article>
  </div>
  <div class="c-bolt-dialog__footer">
    Footer content
  </div>
</dialog>
`;

exports[`Bolt Dialog component dialog theme: xlight 1`] = `
<dialog class="c-bolt-dialog c-bolt-dialog--hidden c-bolt-dialog__width-optimal c-bolt-dialog__spacing-medium t-bolt-xlight">
  <div class="c-bolt-dialog__close">
    <button type="button"
            data-dialog-close
            aria-label="Close dialog"
            class="e-bolt-button e-bolt-button--transparent e-bolt-button--icon-only"
    >
      <span class="e-bolt-button__icon-center"
            aria-hidden="true"
      >
        <bolt-icon name="close"
                   aria-hidden="true"
        >
          <span class="c-bolt-icon c-bolt-icon--close">
            <svg class="c-bolt-icon__icon"
                 xmlns="http://www.w3.org/2000/svg"
                 viewbox="0 0 24 24"
            >
              <path fill="var(--bolt-theme-icon, currentColor)"
                    fill-rule="evenodd"
                    d="M13.4 12l5.3-5.3c.4-.4.4-1 0-1.4a1 1 0 00-1.4 0L12 10.6 6.7 5.3a1 1 0 00-1.4 0 1 1 0 000 1.4l5.3 5.3-5.3 5.3A1 1 0 006 19c.3 0 .5-.1.7-.3l5.3-5.3 5.3 5.3a1 1 0 101.4-1.4L13.4 12z"
              >
              </path>
            </svg>
          </span>
        </bolt-icon>
      </span>
    </button>
  </div>
  <header class="c-bolt-dialog__header">
    <div class="c-bolt-dialog__header-content">
      Header content
    </div>
  </header>
  <div class="c-bolt-dialog__body">
    <article>
      Body content
    </article>
  </div>
  <div class="c-bolt-dialog__footer">
    Footer content
  </div>
</dialog>
`;

exports[`Bolt Dialog component dialog width: auto 1`] = `
<dialog class="c-bolt-dialog c-bolt-dialog--hidden c-bolt-dialog__width-auto c-bolt-dialog__spacing-medium t-bolt-xlight">
  <div class="c-bolt-dialog__close">
    <button type="button"
            data-dialog-close
            aria-label="Close dialog"
            class="e-bolt-button e-bolt-button--transparent e-bolt-button--icon-only"
    >
      <span class="e-bolt-button__icon-center"
            aria-hidden="true"
      >
        <bolt-icon name="close"
                   aria-hidden="true"
        >
          <span class="c-bolt-icon c-bolt-icon--close">
            <svg class="c-bolt-icon__icon"
                 xmlns="http://www.w3.org/2000/svg"
                 viewbox="0 0 24 24"
            >
              <path fill="var(--bolt-theme-icon, currentColor)"
                    fill-rule="evenodd"
                    d="M13.4 12l5.3-5.3c.4-.4.4-1 0-1.4a1 1 0 00-1.4 0L12 10.6 6.7 5.3a1 1 0 00-1.4 0 1 1 0 000 1.4l5.3 5.3-5.3 5.3A1 1 0 006 19c.3 0 .5-.1.7-.3l5.3-5.3 5.3 5.3a1 1 0 101.4-1.4L13.4 12z"
              >
              </path>
            </svg>
          </span>
        </bolt-icon>
      </span>
    </button>
  </div>
  <header class="c-bolt-dialog__header">
    <div class="c-bolt-dialog__header-content">
      Header content
    </div>
  </header>
  <div class="c-bolt-dialog__body">
    <article>
      Body content
    </article>
  </div>
  <div class="c-bolt-dialog__footer">
    Footer content
  </div>
</dialog>
`;

exports[`Bolt Dialog component dialog width: full 1`] = `
<dialog class="c-bolt-dialog c-bolt-dialog--hidden c-bolt-dialog__width-full c-bolt-dialog__spacing-medium t-bolt-xlight">
  <div class="c-bolt-dialog__close">
    <button type="button"
            data-dialog-close
            aria-label="Close dialog"
            class="e-bolt-button e-bolt-button--transparent e-bolt-button--icon-only"
    >
      <span class="e-bolt-button__icon-center"
            aria-hidden="true"
      >
        <bolt-icon name="close"
                   aria-hidden="true"
        >
          <span class="c-bolt-icon c-bolt-icon--close">
            <svg class="c-bolt-icon__icon"
                 xmlns="http://www.w3.org/2000/svg"
                 viewbox="0 0 24 24"
            >
              <path fill="var(--bolt-theme-icon, currentColor)"
                    fill-rule="evenodd"
                    d="M13.4 12l5.3-5.3c.4-.4.4-1 0-1.4a1 1 0 00-1.4 0L12 10.6 6.7 5.3a1 1 0 00-1.4 0 1 1 0 000 1.4l5.3 5.3-5.3 5.3A1 1 0 006 19c.3 0 .5-.1.7-.3l5.3-5.3 5.3 5.3a1 1 0 101.4-1.4L13.4 12z"
              >
              </path>
            </svg>
          </span>
        </bolt-icon>
      </span>
    </button>
  </div>
  <header class="c-bolt-dialog__header">
    <div class="c-bolt-dialog__header-content">
      Header content
    </div>
  </header>
  <div class="c-bolt-dialog__body">
    <article>
      Body content
    </article>
  </div>
  <div class="c-bolt-dialog__footer">
    Footer content
  </div>
</dialog>
`;

exports[`Bolt Dialog component dialog width: optimal 1`] = `
<dialog class="c-bolt-dialog c-bolt-dialog--hidden c-bolt-dialog__width-optimal c-bolt-dialog__spacing-medium t-bolt-xlight">
  <div class="c-bolt-dialog__close">
    <button type="button"
            data-dialog-close
            aria-label="Close dialog"
            class="e-bolt-button e-bolt-button--transparent e-bolt-button--icon-only"
    >
      <span class="e-bolt-button__icon-center"
            aria-hidden="true"
      >
        <bolt-icon name="close"
                   aria-hidden="true"
        >
          <span class="c-bolt-icon c-bolt-icon--close">
            <svg class="c-bolt-icon__icon"
                 xmlns="http://www.w3.org/2000/svg"
                 viewbox="0 0 24 24"
            >
              <path fill="var(--bolt-theme-icon, currentColor)"
                    fill-rule="evenodd"
                    d="M13.4 12l5.3-5.3c.4-.4.4-1 0-1.4a1 1 0 00-1.4 0L12 10.6 6.7 5.3a1 1 0 00-1.4 0 1 1 0 000 1.4l5.3 5.3-5.3 5.3A1 1 0 006 19c.3 0 .5-.1.7-.3l5.3-5.3 5.3 5.3a1 1 0 101.4-1.4L13.4 12z"
              >
              </path>
            </svg>
          </span>
        </bolt-icon>
      </span>
    </button>
  </div>
  <header class="c-bolt-dialog__header">
    <div class="c-bolt-dialog__header-content">
      Header content
    </div>
  </header>
  <div class="c-bolt-dialog__body">
    <article>
      Body content
    </article>
  </div>
  <div class="c-bolt-dialog__footer">
    Footer content
  </div>
</dialog>
`;

exports[`Bolt Dialog component dialog width: regular 1`] = `
<dialog class="c-bolt-dialog c-bolt-dialog--hidden c-bolt-dialog__width-regular c-bolt-dialog__spacing-medium t-bolt-xlight">
  <div class="c-bolt-dialog__close">
    <button type="button"
            data-dialog-close
            aria-label="Close dialog"
            class="e-bolt-button e-bolt-button--transparent e-bolt-button--icon-only"
    >
      <span class="e-bolt-button__icon-center"
            aria-hidden="true"
      >
        <bolt-icon name="close"
                   aria-hidden="true"
        >
          <span class="c-bolt-icon c-bolt-icon--close">
            <svg class="c-bolt-icon__icon"
                 xmlns="http://www.w3.org/2000/svg"
                 viewbox="0 0 24 24"
            >
              <path fill="var(--bolt-theme-icon, currentColor)"
                    fill-rule="evenodd"
                    d="M13.4 12l5.3-5.3c.4-.4.4-1 0-1.4a1 1 0 00-1.4 0L12 10.6 6.7 5.3a1 1 0 00-1.4 0 1 1 0 000 1.4l5.3 5.3-5.3 5.3A1 1 0 006 19c.3 0 .5-.1.7-.3l5.3-5.3 5.3 5.3a1 1 0 101.4-1.4L13.4 12z"
              >
              </path>
            </svg>
          </span>
        </bolt-icon>
      </span>
    </button>
  </div>
  <header class="c-bolt-dialog__header">
    <div class="c-bolt-dialog__header-content">
      Header content
    </div>
  </header>
  <div class="c-bolt-dialog__body">
    <article>
      Body content
    </article>
  </div>
  <div class="c-bolt-dialog__footer">
    Footer content
  </div>
</dialog>
`;
